src/Header.tsx

```diff
@@ -37,10 +41,11 @@ Header.Item = styled.div`
   ${BORDER};
   ${sx};
 `
-Header.Item.displayName = 'Header.Item'
 
-Header.Link = styled.a.attrs(props => {
-  const isReactRouter = typeof props.to === 'string'
+HeaderItem.displayName = 'Header.Item'
+
+const HeaderLink = styled.a.attrs(({to}: StyledHeaderLinkProps) => {
+  const isReactRouter = typeof to === 'string'
```

Comment:
When using `attrs`, we've been passing the props type using the `<>` notation:

```suggestion
const HeaderLink = styled.a.attrs<StyledHeaderLinkProps>(({to}) => {
  const isReactRouter = typeof to === 'string'
```

Line 56-57 should look like this:

```
})<StyledHeaderLinkProps>`
  font-weight: ${get('fontWeights.bold')};
```